src/query/execute_statement.c

```diff
@@ -14581,41 +14604,69 @@ do_prepare_subquery (PARSER_CONTEXT * parser, PT_NODE * stmt)
 {
   int err = NO_ERROR;
   PARSER_CONTEXT context;
-  int i, var_count;
-  PT_NODE *save_next = NULL;
+  int i, sub_idx, var_count;
+  PT_NODE *hv, *save_next = NULL;
+  PT_NODE **host_var_p;
   PT_MISC_TYPE save_flag;
 
   context = *parser;
 
-  context.host_var_count = 0;
   var_count = parser->host_var_count + parser->auto_param_count;
 
   stmt->info.query.flag.subquery_cached = 1;
 
+  context.sub_host_var_index = NULL;
+  context.host_var_count = 0;
+  stmt->sub_host_var_count = 0;
+
   if (var_count > 0)
     {
-      context.sub_host_var_index = (int *) parser_alloc (parser, var_count * sizeof (int));
-      if (context.sub_host_var_index == NULL)
+      host_var_p = (PT_NODE **) malloc (sizeof (PT_NODE *) * var_count);
+      if (host_var_p == NULL)
 	{
-	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, var_count * sizeof (int));
-	  return ER_OUT_OF_VIRTUAL_MEMORY;
+	  goto err_exit;
 	}
-      /* to ininitialize empty index */
-      memset (context.sub_host_var_index, -1, var_count * sizeof (int));
-    }
 
-  /* not to traverse next subquery */
-  save_next = stmt->next;
-  stmt->next = NULL;
+      context.host_variables = (DB_VALUE *) malloc (var_count * sizeof (DB_VALUE));
+      if (context.host_variables == NULL)
+	{
+	  goto err_exit;
+	}
+
+      stmt->sub_host_var_index = (int *) parser_alloc (parser, var_count * sizeof (int));
+      if (stmt->sub_host_var_index == NULL)
+	{
+	  goto err_exit;
+	}
+      memset (host_var_p, NULL, var_count * sizeof (PT_NODE *));
```

Comment:
NULL이라고 써 있으면 컴파일 warning이 발생합니다.
```suggestion
      memset (host_var_p, 0, var_count * sizeof (PT_NODE *));
```

Reply:
네, 감사합니다.